### PR TITLE
hide received tab / add account on watch-only wallet

### PR DESCRIPTION
**Pull request by bugjt**
This PR resolved the issues: 
Closed: #580
Closed: #582
Closed: #578
Closed: #579  
Closed: #581 
#557 

This PR following changes:
- fix crash when click to received tab on watch only wallet
- hide add account button on watch only wallet
- fix crash when click to status proposal detail page
- fix "View on block explorer" button in transaction detail page not work
- fix "copy" button in "View on block explorer" modal not work
- remove tempRightHead in proposal detail page
- always set the watch-only wallet/account's spendable balance to zero
- fix slide navigation buttons and warning on mixer card in dark mode are hard to see
- The confirm password modal to start mixing is stuck spinning

image

image

image

### CODE REVIEW

**Comment by dreacot** [comment]
> hide received tab on watch only wallet

Why are you hiding receive tab, a watch only wallet is able to receive funds

**Comment by bugjt** [comment]
> > hide received tab on watch only wallet
> 
> Why are you hiding receive tab, a watch only wallet is able to receive funds

I have reverted it, and show watch-only wallet on Receive modal

File: ui/page/transaction/transaction_details_page.go [hunk 1]
```diff
@@ -956,6 +947,18 @@ func (pg *TxDetailsPage) pageSections(gtx C, body layout.Widget) D {
 // displayed.
 // Part of the load.Page interface.
 func (pg *TxDetailsPage) HandleUserInteractions(gtx C) {
+
+	for _, item := range pg.moreItems {
```
Comment (JustinBeBoy) [inline ui/page/transaction/transaction_details_page.go:951]: please check `pg.moreItems`,  I don' think it need a slice.

Comment (JustinBeBoy) [reply ui/page/transaction/transaction_details_page.go:951]: please refactor it

**Review by JustinBeBoy — CHANGES_REQUESTED** [review]
I can't copy transaction explorer link in modal and `Got it` button not works

image

- Please check and remove `tempRightHead` in proposal detail page, seem like it not used anywhere

Displaying balance is very confusing, why total balance is 1543.76... DCR but Spendable is 0 DCR? please fix it

image

File: ui/page/transaction/transaction_details_page.go (hunk 1 above)
Comment (bugjt) [reply ui/page/transaction/transaction_details_page.go:951]: yes, I will check it

Comment (bugjt) [reply ui/page/transaction/transaction_details_page.go:951]: If we don’t need more button, I will move “view on block explorer” button to transaction detail page and remove “…” button.

Comment (JustinBeBoy) [reply ui/page/transaction/transaction_details_page.go:951]: @dreacot I think we should move `view on block explorer` button to outside like proposal page...

**Comment by JustinBeBoy** [comment]
@bugjt update your PR description

File: ui/page/transaction/transaction_details_page.go (hunk 1 above)
Comment (bugjt) [reply ui/page/transaction/transaction_details_page.go:951]: Ok, I will do it

**Comment by bugjt** [comment]
It's ready for review

**Comment by dreacot** [comment]
@JustinBeBoy there are several reasons why spendable balance can be zero, e.g watch only wallet, staked tickets, etc

So i don’t see the problem

**Comment by bugjt** [comment]
As I understand it, what Justin said means there is an inconsistency in the send page/modal, destination wallet spendable is 0, while destination account spendable is different from 0.

image

**Comment by JustinBeBoy** [comment]
> @JustinBeBoy there are several reasons why spendable balance can be zero, e.g watch only wallet, staked tickets, etc
> 
> So i don’t see the problem

I see, but it should be consistent across all display pages.

Furthermore `Spendable` can be understood as a spendable balance or a balance in a wallet, but since it is a view-only wallet, the wallet user cannot spend that amount

@bugjt it fine when we show it

**Comment by dreacot** [comment]
> Furthermore Spendable can be understood as a spendable balance or a balance in a wallet

Spendable balance is basically what portion of the wallet balance is available for the user to spend, and in the case of a watch only wallet it should be clearly shown as 0, as users cannot spend from a watch only wallet, they can only receive to it

**Comment by bugjt** [comment]
Ok, This PR is ready for review

File: ui/page/transaction/transaction_details_page.go [hunk 2]
```diff
@@ -898,48 +874,26 @@ func (pg *TxDetailsPage) showbrowserURLModal(copyredirect *cryptomaterial.Clicka
...
-									}
-
-									gtx.Constraints.Min.X = gtx.Constraints.Max.X
-									return pg.Theme.Label(values.TextSize14, pg.moreItems[i].text).Layout(gtx)
-								})
-							})
-						}),
-					)
-				})
-			})
```
Comment (dreacot) [inline ui/page/transaction/transaction_details_page.go:939]: why did you remove the option menu?

what happens on mobile?

**Review by dreacot — CHANGES_REQUESTED** [review]

File: ui/page/transaction/transaction_details_page.go (hunk 1 above)
Comment (bugjt) [reply ui/page/transaction/transaction_details_page.go]: @dreacot Check out this conversation

File: ui/page/transaction/transaction_details_page.go (hunk 2 above)
Comment (bugjt) [reply ui/page/transaction/transaction_details_page.go:939]: please checkout above conversation

Comment (dreacot) [reply ui/page/transaction/transaction_details_page.go:939]: you didn't link any conversation

File: ui/page/transaction/transaction_details_page.go (hunk 1 above)
Comment (dreacot) [reply ui/page/transaction/transaction_details_page.go]: i see this now, my question remains though, what happens to that view on mobile

Comment (dreacot) [reply ui/page/transaction/transaction_details_page.go]: from the comments above , i see.a series of misunderstanding, @JustinBeBoy did not say the more button wasn't needed, he said he doesn't think it should be a slice

Comment (dreacot) [reply ui/page/transaction/transaction_details_page.go]: please revert

Comment (bugjt) [reply ui/page/transaction/transaction_details_page.go:954]: ok, the "View on block explorer" button has been returned to the "more options" button @dreacot

**Comment by bugjt** [comment]
This PR is ready for review

**Review by dreacot — APPROVED** [review]
